### PR TITLE
Add Forta Shares Voting Strategy [forta-shares]

### PR DESCRIPTION
**Pull request by emanuel-sol**
Changes proposed in this pull request:
- Calculates voting power based on the number of EIP 1155 shares for staked FORT

### CODE REVIEW

File: src/strategies/forta-shares/index.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  const response = await fetch("https://api.forta.network/stats/shares", {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(params)
+  });
```
Comment (ChaituVR) [inline src/strategies/forta-shares/index.ts:46]: This API doesn't consider block number? there is a possibility that people can change their VP during a proposal vote period, to give consistent results, better to consider `snapshot` param, 

Or maybe this API returns the same data every time?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/forta-shares/index.ts (hunk 1 above)
Comment (ChaituVR) [reply src/strategies/forta-shares/index.ts:46]: Everything else looks good :)

Comment (emanuel-sol) [reply src/strategies/forta-shares/index.ts:46]: This API returns a list of all the shares ever owned by a particular address. We use the contract queries to determine the voting power of these at execution time (which i'm guessing is when a client signs in with a wallet to vote on our proposal? still a tad confused about how snapshot works under the hood)

Comment (ChaituVR) [reply src/strategies/forta-shares/index.ts:46]: Hello  again @emanuel-sol So snapshot doesn't really consider time of user vote, `snapshot` block number is what decides the voting power of the user for the proposal, (this block number is taken at time of creation) 
For example if you consider erc20-balance-of strategy, voting power here is decided by balance held in a user's wallet the proposal's snapshot block number

so the API should ideally return the shares owned at a particular block time

Comment (stevenlanders) [reply src/strategies/forta-shares/index.ts:46]: @ChaituVR the API returns all tokens ever owned up to the current second, but then we use the _snapshot_ block time to get the balance.  It starts a superset, then is reduced by the snapshot balances (will receive 0 balances for any tokens no longer owned).  This reduction is why the API doesn't influence the voting power in any problematic way.

**Review by ChaituVR — APPROVED** [review]

File: src/strategies/forta-shares/index.ts (hunk 1 above)
Comment (ChaituVR) [reply src/strategies/forta-shares/index.ts:46]: Thanks for explaining @stevenlanders :)

**Pull request merged**